src/token/TokenRejectTransaction.js

```diff
@@ -0,0 +1,295 @@
+ */
+export default class TokenRejectTransaction extends Transaction {
+    /**
+     *
+     * @param {object} [props]
+     * @param {AccountId | string | null} [props.owner]
+     * @param {NftId[]} [props.nftIds]
+     * @param {TokenId[]} [props.tokenIds]
+     */
+    constructor(props = {}) {
+        super();
+
+        /**
+         * @private
+         * @type {?AccountId}
+         */
+        this._owner = null;
+
+        if (props.owner != null) {
+            this.setOwnerId(props.owner);
+        }
+
+        /**
+         * @private
+         * @type {TokenId[]}
+         */
+        this._tokenIds = [];
+
+        /**
+         * @private
+         * @type {NftId[]}
+         */
+        this._nftIds = [];
+
+        if (props.tokenIds != null) {
+            this._tokenIds = props.tokenIds;
+        }
+
+        if (props.nftIds != null) {
+            this._nftIds = props.nftIds;
+        }
+    }
+
+    /**
+     * @internal
+     * @param {HashgraphProto.proto.ITransaction[]} transactions
+     * @param {HashgraphProto.proto.ISignedTransaction[]} signedTransactions
+     * @param {TransactionId[]} transactionIds
+     * @param {AccountId[]} nodeIds
+     * @param {HashgraphProto.proto.ITransactionBody[]} bodies
+     * @returns {TokenRejectTransaction}
+     */
+    static _fromProtobuf(
+        transactions,
+        signedTransactions,
+        transactionIds,
+        nodeIds,
+        bodies,
+    ) {
+        const body = bodies[0];
+        const rejectToken =
+            /** @type {HashgraphProto.proto.ITokenRejectTransactionBody} */ (
+                body.tokenReject
+            );
+
+        const tokenIds = rejectToken.rejections?.map((rejection) =>
+            TokenReference._fromProtobuf(rejection),
+        );
+        const ftIds = tokenIds
+            ?.filter((token) => token.tokenIdentifier === "fungibleToken")
+            .map(({ fungibleToken }) => {
+                if (fungibleToken == null) {
+                    throw new Error("Fungible Token cannot be null");
+                }
+                return fungibleToken;
+            });
+
+        const nftIds = tokenIds
+            ?.filter((token) => token.tokenIdentifier === "nft")
+            .map(({ nft }) => {
+                if (nft == null) {
+                    throw new Error("Nft cannot be null");
+                }
+                return nft;
+            });
+
+        return Transaction._fromProtobufTransactions(
+            new TokenRejectTransaction({
+                owner:
+                    rejectToken.owner != null
+                        ? AccountId._fromProtobuf(rejectToken.owner)
+                        : undefined,
+
+                tokenIds: ftIds,
+                nftIds: nftIds,
+            }),
+            transactions,
+            signedTransactions,
+            transactionIds,
+            nodeIds,
+            bodies,
+        );
+    }
+
+    /**
+     * @returns {TokenId[] | string[]}
```

Comment:
The result will be `TokenId[]` always as it's declared in the constructor.

```suggestion
     * @returns {?TokenId[]}
```